frontend/www/js/omegaup/components/login/Signup.vue

```diff
@@ -106,31 +128,79 @@
 </template>
 
 <script lang="ts">
-import { Vue, Component, Prop } from 'vue-property-decorator';
+import { Vue, Component, Prop, Watch } from 'vue-property-decorator';
 import omegaup_Markdown from '../Markdown.vue';
 import T from '../../lang';
-
+import * as time from '../../time';
+import DatePicker from '../DatePicker.vue';
+import * as ui from '../../ui';
 @Component({
   components: {
     'omegaup-markdown': omegaup_Markdown,
+    'omegaup-datepicker': DatePicker,
   },
 })
 export default class Signup extends Vue {
   @Prop() validateRecaptcha!: boolean;
-
   T = T;
   username: string = '';
-  email: string = '';
+  email: null | string = null;
+  parentEmail: null | string = null;
   password: string = '';
   passwordConfirmation: string = '';
   recaptchaResponse: string = '';
+  birthDate: null | Date = null;
+  privacyPolicyAccepted = false;
+  get loginEmailDescriptionText(): string {
+    if (!this.userAge) {
+      return T.loginEmail;
+    }
+    return this.userAge > 13 ? T.loginEmail : T.loginEmailParent;
+  }
+  get userAge(): number | null {
+    if (this.birthDate === null) {
+      return null;
+    }
+    return time.getDifferenceInCalendarYears(this.birthDate);
+  }
```

Comment:
it's probably better to consolidate the test of the user being under 13 into a single function that can be used everywhere because there is a bug in all the age checks: the opposite of `this.userAge < 13` (the definition of U13) is not `this.userAge > 13`, but `this.userAge >= 13`! this means that this form is checking for U14, not U13.

```suggestion
  get isU13(): boolean {
    if (this.birthDate === null) {
      // Most users are not going to be U13. So until they fill out their
      // birthdate, assume they aren't so that they can see the default form.
      return false;
    }
    return time.getDifferenceInCalendarYears(this.birthDate) < 13;
  }
```

this way all the `this.userAge > 13` can be replaced with `!this.isU13`